pkg/reconciler/ingress/resources/virtual_service.go

```diff
@@ -0,0 +1,257 @@
+/*
+Copyright 2019 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package resources
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/apimachinery/pkg/util/sets"
+
+	istiov1alpha1 "github.com/knative/pkg/apis/istio/common/v1alpha1"
+	"github.com/knative/pkg/apis/istio/v1alpha3"
+	"github.com/knative/pkg/kmeta"
+	"github.com/knative/pkg/system"
+	"github.com/knative/serving/pkg/apis/networking"
+	"github.com/knative/serving/pkg/apis/networking/v1alpha1"
+	"github.com/knative/serving/pkg/apis/serving"
+	"github.com/knative/serving/pkg/network"
+	"github.com/knative/serving/pkg/reconciler/ingress/resources/names"
+	"github.com/knative/serving/pkg/resources"
+)
+
+// VirtualServiceNamespace gives the namespace of the child
+// VirtualServices for a given Ingress.
+func VirtualServiceNamespace(_ *v1alpha1.Ingress) string {
+	return system.Namespace()
```

Comment:
This should use the namespace of the Ingress param?

Reply:
Yes.   I'd like to do that in the second PR though.

Reply:
Done.